### PR TITLE
Tekton example for running ORT on Kubernetes

### PR DESCRIPTION
**Pull request by cgrotz**
This PR contains an example how to run ORT on K8S clusters using Tekton. I would highly appreciate some feedback!

### CODE REVIEW

**Comment by sschuberth** [comment]
Thanks a lot for this @cgrotz! Before starting the real review, could you please get rid of the merge commit in the PR, and rather *rebase* onto latest `main`?

**Comment by cgrotz** [comment]
> Thanks a lot for this @cgrotz! Before starting the real review, could you please get rid of the merge commit in the PR, and rather _rebase_ onto latest `main`?

Sure, the GitHub `Fetch Upstream` feature got me there. Should be fixed now.

File: integrations/tekton/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Tekton Pipeline Example
+
+This example pipeline allows you to easily run an ORT scan on [Tekton](https://tekton.dev/), some knowledge about Tekton will help to understand this example pipeline.. This example pipeline works best on [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine).
```
Comment (sschuberth) [inline integrations/tekton/README.md:3]: - Maybe just "run ORT on"?
- Start a new sentence from "Some knowledge..."?
- s/pipeline../pipeline./
- It is really true that this "works best" on GKE, or has it only been tested on GKE? If indeed the former, could you briefly add why?

File: integrations/tekton/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+## Getting Started
+To run the pipeline, you will need the following things:
```
Comment (sschuberth) [inline integrations/tekton/README.md:6]: Nit: Please add a blank line after the title for readability, like above.

File: integrations/tekton/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+* A Kubernetes Cluster with Tekton installed (follow the [installation instructions](https://tekton.dev/docs/pipelines/install/))
```
Comment (sschuberth) [inline integrations/tekton/README.md:7]: - s/Cluster/cluster
- In general, please also end sentences from bullet lists with a dot (also in lists below).

File: integrations/tekton/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+* You will need to install the [git-clone](https://hub.tekton.dev/tekton/task/git-clone) and [gcs-upload](https://hub.tekton.dev/tekton/task/gcs-upload) tasks
+* A [Google Cloud Storage](https://cloud.google.com/products/storage) bucket
+
+The pipeline definition is in the `ort-pipeline.yaml` you will need to replace `ort:latest` with the full image name e.g. `gcr.io/<project_id>/ort:latest`.
```
Comment (sschuberth) [inline integrations/tekton/README.md:11]: Maybe: "The pipeline is defined in `ort-pipeline.yaml`. You will..."

Comma before "e.g.".

File: integrations/tekton/README.md (hunk 1 above)
Comment (sschuberth) [reply integrations/tekton/README.md:3]: Also, we should probably explain acronyms once before using them later, like

[Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine) (GKE)

File: integrations/tekton/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# Tekton Pipeline Example
+
+This example pipeline allows you to easily run an ORT scan on [Tekton](https://tekton.dev/), some knowledge about Tekton will help to understand this example pipeline.. This example pipeline works best on [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine).
+
+## Getting Started
+To run the pipeline, you will need the following things:
+* A Kubernetes Cluster with Tekton installed (follow the [installation instructions](https://tekton.dev/docs/pipelines/install/))
+* You will need to install the [git-clone](https://hub.tekton.dev/tekton/task/git-clone) and [gcs-upload](https://hub.tekton.dev/tekton/task/gcs-upload) tasks
+* A [Google Cloud Storage](https://cloud.google.com/products/storage) bucket
```
Comment (sschuberth) [inline integrations/tekton/README.md:9]: Also explain acronym here for later use: [Google Cloud Storage](https://cloud.google.com/products/storage) (GCS)

File: integrations/tekton/README.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+The pipeline definition is in the `ort-pipeline.yaml` you will need to replace `ort:latest` with the full image name e.g. `gcr.io/<project_id>/ort:latest`.
+
+The pipeline takes a few inputs:
+* `shared-data` is a workspace and will need a volume, it's used for checking out the source code repo, and storing the temporary artifacts
+* `gcs-credentials` is a workspace and will need a volume. On GKE you can use Workload Identities to allow the KSA to access the GCS bucket for uploading the results. (If you are not using Workload Idenity, find more information how to configure credentials in the [gcs-upload](https://hub.tekton.dev/tekton/task/gcs-upload) documentation)
```
Comment (sschuberth) [inline integrations/tekton/README.md:15]: What's KSA?

File: integrations/tekton/README.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+* `repo-url` is a param, and takes the path to the Git repo that should be scanned
```
Comment (sschuberth) [inline integrations/tekton/README.md:16]: s/param/parameter/

(Unless "param" really is a coined term here. Same below.)

File: integrations/tekton/README.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+* `result-bucket-path` is a param, and takes the path to GCS where the result report should be stored e.g. gs://<gcs-bucket>/<sub_path>
+
+In order to get started call `kubectl apply -f ort-pipeline.yaml` to create the pipeline on your Kubernetes cluster. Now, if you want to trigger a pipeline run, you need to create a `PipelineRun` resource. In order to trigger a run, for example for the [Simple Maven example](https://github.com/MarcelBochtler/maven-simple) you can call `kubectl apply -f simple-maven.yaml` (or `kubectl apply -f mime-types.yaml` for the [Mime Types repo](https://github.com/jshttp/mime-types)).
```
Comment (sschuberth) [inline integrations/tekton/README.md:19]: Comma before "you can call".

File: integrations/tekton/README.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+![Example run of Simple Maven](imgs/simple-maven.png)
```
Comment (sschuberth) [inline integrations/tekton/README.md:21]: Let's name the directory "images".

File: integrations/tekton/README.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+![Example run of Mime Types](imgs/mime-types.png)
+
+## Next Steps
+You probably don't want to trigger each execution via kubectl. Checkout the [Tekton documentation](https://tekton.dev/docs/triggers/) to learn how you can trigger a PipelineRun via EventListeneres.
```
Comment (sschuberth) [inline integrations/tekton/README.md:26]: - s/kubectl/`kubectl`/
- s/Checkout/Check out/
- Probably "PipelineRun" and "EventListeneres" should be formatted to stand out, maybe like *PipelineRun* and *EventListeneres*.

File: integrations/tekton/README.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+If you aren't working on a public GitHub repo you should also checkout how to work with a [private repo](https://hub.tekton.dev/tekton/task/git-clone).
```
Comment (sschuberth) [inline integrations/tekton/README.md:28]: Spell out "repo" as "repository".

File: integrations/tekton/mime-types.yaml [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+          requests:
+            storage: 1Gi
+  - name: ort-data-dir
+    persistentVolumeClaim:
+      claimName: ort-data-dir
+  params:
+  - name: repo-url
+    value: https://github.com/jshttp/mime-types.git
+  - name: result-bucket-path
+    value: gs://ort-tekton/reports
```
Comment (sschuberth) [inline integrations/tekton/mime-types.yaml:33]: Missing final newline.

File: integrations/tekton/simple-maven.yaml [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+          requests:
+            storage: 1Gi
+  - name: ort-data-dir
+    persistentVolumeClaim:
+      claimName: ort-data-dir
+  params:
+  - name: repo-url
+    value: https://github.com/MarcelBochtler/maven-simple.git
+  - name: result-bucket-path
+    value: gs://ort-tekton/reports
```
Comment (sschuberth) [inline integrations/tekton/simple-maven.yaml:33]: Missing final newline.

**Review by sschuberth — CHANGES_REQUESTED** [review]
Just a bunch of docs nits 😁

**Comment by sschuberth** [comment]
Thanks @cgrotz for the updates! Please squash all commits into one, as we prefer to have review comments addressed in the commits they belong to.

Also, we follow Git's convention for writing commit messages, so please use [imperative mood](https://github.com/oss-review-toolkit/.github/blob/main/CONTRIBUTING.md#git-history), i.e. instead of "Created ..." say "Create ...".

**Comment by sschuberth** [comment]
Also the REUSE check fails, because your newly added files need to be explicitly licensed via entries in the `.reuse/dep5` file.

**Review by sschuberth — APPROVED** [review]

**Comment by sschuberth** [comment]
Merging despite the unrelated `ConanFunTest` failure.

**Pull request merged**